internal/project/common/gh_workflow.go

```diff
@@ -158,6 +159,10 @@ func (gh *GHWorkflow) CompileGitHubWorkflow(o *ghworkflow.Output) error {
 						SetWith("path", step.ArtifactStep.ArtifactPath+"\n"+strings.Join(step.ArtifactStep.AdditionalArtifacts, "\n")).
 						SetWith("retention-days", "5")
 
+					if step.ArtifactStep.RetentionDays != "" {
+						saveArtifactsStep.SetWith("retention-days", step.ArtifactStep.RetentionDays)
```

Comment:
Minor nit: here and below I think it's best to use something like
```suggestion
					if retentionDays := step.ArtifactStep.RetentionDays; retentionDays != "" {
						saveArtifactsStep.SetWith("retention-days", retentionDays)
```

IMHO it's just easier to reason about.